src/topn.c

```diff
@@ -0,0 +1,126 @@
+#include "data.table.h"
+
+static inline void swap(int *a, int *b) { int tmp=*a; *a=*b; *b=tmp; }
+
+static inline bool icmp(const int *x, const int a, const int b, const bool min, const bool nalast) {
+  if (x[a]==x[b]) return(a > b);
+  if (x[a]==NA_INTEGER) return(nalast);
+  if (x[b]==NA_INTEGER) return(!nalast);
+  return(min ? x[a] < x[b] : x[a] > x[b]);
+}
+static inline bool dcmp(const double *x, const int a, const int b, const bool min, const bool nalast) {
+  if (x[a]==x[b] || isnan(x[a]) && isnan(x[b])) return(a > b);
```

Comment:
`||` vs. `&&` operator precedence is hard to remember, use parens to make the grouping clearer